src/scripts/main.js

```diff
@@ -1,3 +1,217 @@
 'use strict';
 
 // write your code here
+const button = document.querySelector('.button');
+const messeageStart = document.querySelector('.message-start');
+const messageWin = document.querySelector('.message-win');
+const messageLose = document.querySelector('.message-lose');
+const gameScore = document.querySelector('.game-score');
+const rows = document.querySelectorAll('.field-row');
+let score = 0;
+let gameField = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+function setGameField() {
+  for (let row = 0; row < 4; row++) {
+    for (let cell = 0; cell < 4; cell++) {
+      rows[row].children[cell].textContent = gameField[row][cell] || '';
+
+      rows[row].children[cell].className
+        = `field-cell field-cell--${gameField[row][cell]}`;
+
+      if (gameField[row][cell] === 2048) {
+        messageWin.classList.remove('hidden');
+      };
+    }
+  }
+};
+
+function hasEmptyCell() {
+  for (let row = 0; row < 4; row++) {
+    for (let cell = 0; cell < 4; cell++) {
+      if (gameField[row][cell] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function canMove() {
+  for (let r = 0; r < 4; r++) {
+    for (let c = 0; c < 3; c++) {
+      if (gameField[c][r] === gameField[c + 1][r]
+        || gameField[r][c] === gameField[r][c + 1]) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function setTwoOrFour() {
+  canMove();
+
+  if (!canMove() && !hasEmptyCell()) {
+    messageLose.classList.remove('hidden');
+
+    return;
+  }
+
+  const row = Math.floor(Math.random() * 4);
+  const cell = Math.floor(Math.random() * 4);
+
+  if (hasEmptyCell()) {
+    (gameField[row][cell] === 0)
+      ? gameField[row][cell] = Math.floor(Math.random() >= 0.9 ? 4 : 2)
+      : setTwoOrFour();
```

Comment:
It's better to avoid recursion here as you repeat check for the ability to move and presence of empty cells even though the game field hasn't changed. You should only redeclare random row and cell for them to be in empty field.